mca/src/api/app_manager.rs

```diff
@@ -0,0 +1,58 @@
+
+use std::{ collections::HashMap, sync::{Arc, Mutex}};
+
+use tonic::{metadata::MetadataValue, Request, Status};
+
+use crate::common::utils::get_id;
+
+use super::app_base::{AppBase, Role};
+
+use lazy_static::lazy_static;
+
+pub struct AppManager {
+    pub app_list:HashMap<usize, Arc<Mutex<AppBase>>>,
+    pub n_apps: u8
+}
+
+impl AppManager {
+    pub fn new() -> Self {
+        AppManager { app_list: HashMap::new(), n_apps:0 }
+    }
+
+    fn register(&mut self, name: String, addr:String, role: Role) -> Arc<Mutex<AppBase>> {
+        let app_point = Arc::new(Mutex::new(AppBase::new(name, addr, role)));
```

Comment:
Why `AppBase` is wrapped with `Mutex`? I know `Mutex` is for multi thread environment. Is app planning to use multi thread? Let's discuss at meeting.

Reply:
This is not used, yet. At first, I planned to assign the struct `AppBase` to each app, and each `AppBase` is referred to in two types of threads:
1. Change Role in AppManager: In this case, `AppBase`(precisely, member `Role`) is written by AppManager
2. Check Author in AppManager: In this case, `AppBase`(precisely, member `Role`) is read by AppManager

I don't think these steps occur at the same time, but I think it should have a mutex. But I cannot sure, haha;